File: frontend/src/types/analysis.ts
export interface ScoreBreakdown {
  skill_match: number;
  keyword_match: number;
  semantic_similarity: number;
  experience_relevance: number;
  final_score: number;
}

export interface ParsedSections {
  skills: string[];
  experience: string[];
  education: string[];
  certifications: string[];
  projects: string[];
  keyword_hits: string[];
}

export interface Suggestion {
  category: string;
  message: string;
}

export interface AnalysisResponse {
  analysis_id: string;
  ats_score: number;
  score_breakdown: ScoreBreakdown;
  matched_skills: string[];
  missing_skills: string[];
  semantic_similarity: number;
  parsed_sections: ParsedSections;
  suggestions: Suggestion[];
  report_url: string;
}

export interface AnalyzeRequest {
  resumeFile: File;
  jdText?: string;
  jdFile?: File | null;
}

File: backend/app/schemas/analysis.py
from __future__ import annotations

from pydantic import BaseModel, Field


class ScoreBreakdown(BaseModel):
    skill_match: float = Field(ge=0, le=100)
    keyword_match: float = Field(ge=0, le=100)
    semantic_similarity: float = Field(ge=0, le=100)
    experience_relevance: float = Field(ge=0, le=100)
    final_score: float = Field(ge=0, le=100)


class ParsedSections(BaseModel):
    skills: list[str]
    experience: list[str]
    education: list[str]
    certifications: list[str]
    projects: list[str]
    keyword_hits: list[str]


class Suggestion(BaseModel):
    category: str
    message: str


class AnalysisResponse(BaseModel):
    analysis_id: str
    ats_score: float = Field(ge=0, le=100)
    score_breakdown: ScoreBreakdown
    matched_skills: list[str]
    missing_skills: list[str]
    semantic_similarity: float = Field(ge=0, le=100)
    parsed_sections: ParsedSections
    suggestions: list[Suggestion]
    report_url: str


class AnalysisSummaryResponse(BaseModel):
    analysis_id: str
    resume_filename: str
    jd_source: str
    ats_score: float
    score_breakdown: ScoreBreakdown
    report_url: str
    analysis_json_path: str
    created_at: str


class HealthResponse(BaseModel):
    status: str
    service: str

